sysutils/btop/Portfile

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        aristocratos btop 7e50b03e391a346b2b6ce92be332c58d91f75a00
+version             20230615
+
+fetch.type          git
+
+post-fetch {
+    system -W ${worksrcpath} "git submodule update --init"
+}
+
+description         A monitor of resources
+
+long_description    Resource monitor that shows usage and stats for processor, memory, disks, network and processes.
+
+categories          sysutils
+license             Apache-2
+maintainers         {samasaur.com:sam @samasaur1} openmaintainer
+
+use_configure       no
+depends_build       port:coreutils port:gmake port:clang-16
+destroot.args       PREFIX=${prefix}
+build.args          CXX=clang++-mp-16
```

Comment:
Don't manually set the compiler like this. If you require a specific language standard (such as C++20) specify that (e.g. `compiler.cxx_standard 2020`) and let MacPorts pick a compatible compiler. Using the makefile portgroup as @i0ntempest mentioned would be a prerequisite to getting that to work with a port like this that doesn't use a standard configure phase.

Reply:
And if it is C++20 that you require, MacPorts base 2.8.1 doesn't actually know what the requirements of C++20 are yet (see https://trac.macports.org/ticket/62677) so until there is a new MacPorts base release you'll have to add that information to the port:
```
PortGroup compiler_blacklist_versions 1.0
```
```
compiler.blacklist-append {clang < 1400}
```